lib/evmone/baseline.cpp

```diff
@@ -12,16 +13,21 @@
 
 namespace evmone::baseline
 {
-CodeAnalysis analyze(const uint8_t* code, size_t code_size)
+namespace
+{
+// TODO consider splitting to analyze_legacy and analyze_eof1
+CodeAnalysis analyze_pre_eof2(const uint8_t* code, size_t code_size, const EOF1Header& header)
 {
     // To find if op is any PUSH opcode (OP_PUSH1 <= op <= OP_PUSH32)
     // it can be noticed that OP_PUSH32 is INT8_MAX (0x7f) therefore
     // static_cast<int8_t>(op) <= OP_PUSH32 is always true and can be skipped.
     static_assert(OP_PUSH32 == std::numeric_limits<int8_t>::max());
 
-    CodeAnalysis::JumpdestMap map(code_size);  // Allocate and init bitmap with zeros.
-    size_t i = 0;
-    while (i < code_size)
+    const auto code_begin = header.code_begin();
+    const auto code_end = header.code_end(code_size);
+    CodeAnalysis::JumpdestMap map(code_end);  // Allocate and init bitmap with zeros.
```

Comment:
Why `code_end` not keep `code_size`?

Reply:
A small optimization - there're no jumpdests after code section end, so no point in storing those 0 bits in the map.